src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -45,22 +45,34 @@ namespace
 
 #if defined(NATIVE_LIBS_EMBEDDED)
     extern "C" const void* CompressionResolveDllImport(const char* name);
-
-#if defined(_WIN32)
-#define COMPRESSION_DLL_NAME "System.IO.Compression.Native"
-#else
-#define COMPRESSION_DLL_NAME "libSystem.IO.Compression.Native"
-#endif
+    extern "C" const void* SecurityResolveDllImport(const char* name);
+    extern "C" const void* SystemResolveDllImport(const char* name);
 
     // pinvoke_override:
     // Check if given function belongs to one of statically linked libraries and return a pointer if found.
     const void* STDMETHODCALLTYPE pinvoke_override(const char* libraryName, const char* entrypointName)
     {
-        if (strcmp(libraryName, COMPRESSION_DLL_NAME) == 0)
+#if defined(_WIN32)
+        if (strcmp(libraryName, "System.IO.Compression.Native") == 0)
         {
             return CompressionResolveDllImport(entrypointName);
         }
+#else
+        if (strcmp(libraryName, "libSystem.IO.Compression.Native") == 0)
+        {
+            return CompressionResolveDllImport(entrypointName);
+        }
```

Comment:
nit: can be de-duplicated via a simple macro, e.g.:

```c++
#if defined(_WIN32)
#define LIBNAME(x) x
#else
#define LIBNAME(x) "lib" x
#endif

..

        if (strcmp(libraryName, LIBNAME("System.IO.Compression.Native")) == 0)
        {
            return CompressionResolveDllImport(entrypointName);
        }
```

Reply:
Because there are other platform differences - Windows vs. Unix vs. OSX, there is still a need for `#ifdef`, so I think extra macro would not make it more compact or easier to manage.